yapper/src/tx.rs

```diff
@@ -0,0 +1,69 @@
+use reqwest::{Client, ClientBuilder, Response, Url};
+use std::time::Duration;
+use timeboost::types::BundleVariant;
+use timeboost_crypto::DecryptionScheme;
+use timeboost_crypto::traits::threshold_enc::ThresholdEncScheme;
+
+use anyhow::{Context, Result};
+use cliquenet::Address;
+use timeboost_utils::load_generation::{make_bundle, tps_to_millis};
+use tokio::time::interval;
+use tracing::{error, warn};
+
+async fn send_transaction(
+    client: Client,
+    addr: &Address,
+    pubkey: &<DecryptionScheme as ThresholdEncScheme>::PublicKey,
+) -> Result<Response> {
+    let bundle = make_bundle(pubkey)?;
+
+    match bundle {
+        BundleVariant::Regular(bundle) => {
+            let submision_url = Url::parse(&format!("http://{}/v0/submit-regular", addr))
+                .context(format!("parsing {} into a url", addr))?;
```

Comment:
Perhaps the address parsing can happen only once and not for every bundle? Also, to avoid creating the potentially unecessary error string there is `with_context`:
```suggestion
            let submision_url = Url::parse(&format!("http://{}/v0/submit-regular", addr))
                .with_context(|| format!("parsing {} into a url", addr))?;
```

Reply:
https://github.com/EspressoSystems/timeboost/pull/304/commits/1b873f4e5e1c698176fb57b6664534387abe1619